src/main.py

```diff
@@ -78,6 +79,42 @@ def load_with_usb():
     TO_RUN_IN_MAIN = start_kolibri_with_usb
 
 
+def run_js(js):
+    PythonActivity.mWebView.evaluateJavascript(js, None)
+
+
+run_js = Runnable(run_js)
+
+
+def is_usb_connected():
+    """
+    Check if the KOLIBRI_HOME db file is reachable.
+
+    This only works after the user has granted permissions correctly. In other
+    case it always return False.
+    """
+
+    key_uris = get_endless_key_uris()
+    if not key_uris:
+        return False
+    try:
+        # Check if the USB is connected
+        stat_file(key_uris.get("db"))
+        run_js("setHasUSB(true)")
+        return True
+    except FileNotFoundError:
+        run_js("setHasUSB(false)")
+        return False
+
+
+def check_if_usb_connected_loop():
+    if not is_usb_connected():
+        # Check again later
+        return True
+
+    return False
```

Comment:
I have added a log line here and verified that this is called way too often, every few milliseconds. Please add a waiter. Maybe `time.sleep(0.5)` works here like in the main while loop. Or maybe is better to increase the 0.05 sleep in the main loop for all the "to run in main" functions.

Nit: a better name for this function could be `wait_until_usb_is_connected()`.

Nit: the logic can be simplified to:
```suggestion
def wait_until_usb_is_connected():
    repeat = not is_usb_connected()
    return repeat
```